modules/schedules/schedules.go

```diff
@@ -51,25 +59,92 @@ type Algs2_Request struct {
 }
 
 type Algs1_Request struct {
-	Year       string                 `json:"year"`
-	Term       string                 `json:"term"`
-	Users      []users.User           `json:"users"`
-	Courses    []courses.Course       `json:"courses"`
-	Classrooms []classrooms.Classroom `json:"classrooms"`
-	Capacity   []Capacity             `json:"capacity"`
+	Year       string                  `json:"year"`
+	Term       string                  `json:"term"`
+	Users      []users.User            `json:"users"`
+	Courses    []CoursesWithCapacities `json:"courses"`
+	Classrooms []classrooms.Classroom  `json:"classrooms"`
 }
 
 type Estimate struct {
 	Course   string `json:"course"`
 	Estimate int    `json:"estimate"`
 }
 
+type CoursesWithCapacities struct {
+	Course        string     `json:"course" bson:"course"`
+	Peng          bool       `json:"peng" bson:"peng"`
+	Prerequisites [][]string `json:"prerequisites" bson:"prerequisites"`
+	CoRequisites  []string   `json:"corequisites" bson:"corequisites"`
+	Pre_enroll    int        `json:"pre_enroll" bson:"pre_enroll"`
+	Min_enroll    int        `json:"min_enroll" bson:"min_enroll"`
+	Hours         [3]int     `json:"hours" bson:"hours"`
+}
+
 type Capacity struct {
-	Estimate []Estimate `json:"estimates"`
+	Estimates []Estimate `json:"estimates"`
+}
+
+func make_array_of_courses_with_capacities(term_courses []courses.Course, pred_capacities Capacity) []CoursesWithCapacities {
```

Comment:
I think these two function names can be cleaned up a bit to be createCoursesArray and createScheduleJSON or something similar just to follow the same convention as the rest of the project and for best practice. But yeah the everything else is looking goof I think just being nitpicky.

Reply:
I updated the functions in question and used the function names you suggested. Good catch and thank you for ensuring consistent style